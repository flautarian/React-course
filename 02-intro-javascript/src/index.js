console.log("Hello world!!!")

const dice = {
    faces: 20,
    getRandomValue: function(){
        // 👇️ get number between min (inclusive) and max (inclusive)
        return Math.floor(Math.random() * (this.faces+ 1));
    }
}

const diceResult =`${dice.getRandomValue()}`;
console.log(diceResult);