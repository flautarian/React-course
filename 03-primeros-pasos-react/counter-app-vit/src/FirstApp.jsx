import React from 'react'
import { Fragment } from 'react';

function getPerson() {
    let name = prompt("cual es tu nombre?") || "leonidas";
    let surname = prompt("cual es tu apellido?") || "deuteronomus";
    let job = prompt("cual es tu oficio?") || "espartano";
    return {
        name: name,
        surname: surname,
        job: job
    };
}

let person = getPerson();

export const FirstApp = () => {
    return (
        <Fragment>
            <p>Nombre:</p>
            <p>-Mi nombre es {person.name}</p>
            <p>-Mi apellido es {person.surname}</p>
            <p>-Y mi trabajo es {person.job}</p>
        </Fragment>
    )
}
