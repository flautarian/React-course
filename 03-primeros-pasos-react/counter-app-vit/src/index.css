html, body {
    background-color: #21232A;
    color: white;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1em;
    padding: 70px;
  }
  
  code {
    font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
      monospace;
  }
  
button {
  padding: 15px;
  margin: 5px;
  font-size: 1.3em;
}