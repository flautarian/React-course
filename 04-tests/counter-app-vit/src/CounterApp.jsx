import React from 'react'
import { Fragment } from 'react'
import PropTypes from 'prop-types';
import { useState } from 'react';


export const CounterApp = ({value}) => {

    const [ counter, setCounter ] = useState( value );

    const addOne = ( event ) => setCounter((c) => c + 1 );

    const substractOne = ( event ) => setCounter((c) => c - 1 );

    const resetValue = ( event ) => setCounter((c) => c = value);
    
    
    return (
        <Fragment>
            <h1>CounterApp</h1>
            <h2>Valor: { counter } </h2>
            <button onClick={ (event) => addOne( event ) }>+1</button>
            <button onClick={ (event) => substractOne( event ) }>-1</button>
            <button onClick={ (event) => resetValue( event ) }>Reset</button>
        </Fragment>
    )
}

CounterApp.propTypes ={
    value: PropTypes.number.isRequired
}