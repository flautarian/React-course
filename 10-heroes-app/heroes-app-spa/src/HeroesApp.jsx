import React from 'react'
import { Fragment } from 'react'
import { AuthProvider } from './auth'
import { AppRouter } from './router/AppRouter'

export const HeroesApp = () => {
  return (
    <AuthProvider>
        <AppRouter/>
    </AuthProvider>
  )
}
