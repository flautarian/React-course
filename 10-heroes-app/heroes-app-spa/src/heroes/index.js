export * from './pages';
export * from './routes/HeroesRoutes';
export * from './helpers/getHeroById';
export * from './helpers/getHeroesByPublisher';