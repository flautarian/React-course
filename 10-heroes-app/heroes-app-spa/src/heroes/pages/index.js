export * from './MarvelPage';
export * from './DcPage';
export * from './SearchHeroPage';
export * from './HeroPage';