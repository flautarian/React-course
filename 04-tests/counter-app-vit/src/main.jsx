import React from "react";
import ReactDOM from "react-dom/client";
import { Title } from './Title';
import { FirstApp } from "./FirstApp";
import './index.css';
import { CounterApp } from "./CounterApp";

let valueCounterApp = 10;

ReactDOM.createRoot( document.getElementById('root')).render(
  <React.StrictMode>
    <Title title="Hola!, bienvenido a mi App" subtitle="subtitulo"/>
    <FirstApp/>
    <CounterApp value={valueCounterApp}/>
  </React.StrictMode>
)