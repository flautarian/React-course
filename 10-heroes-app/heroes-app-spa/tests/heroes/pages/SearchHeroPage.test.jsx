import { fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { SearchHeroPage } from '../../../src/heroes';

const mockedUseNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockedUseNavigate,
}));



describe('SearchHeroPage test', () => {


    beforeEach(() => jest.clearAllMocks() );

    
    test('should show default correctly', () => {
        
        const { container } =render(
            <MemoryRouter>
                <SearchHeroPage />
            </MemoryRouter>
        );
        expect( container ).toMatchSnapshot();
        
    });

    test('should show batman and inputs', () => {
        
        render(
            <MemoryRouter initialEntries={['/search?q=batman']}>
                <SearchHeroPage />
            </MemoryRouter>
        );
        
        const input = screen.getByRole('textbox');
        expect( input.value ).toBe('batman');
        
        const img = screen.getByRole('img');
        expect( img.src ).toContain('/assets/heroes/dc-batman.jpg');

        const alert = screen.getByLabelText('alert-danger');
        expect( alert.style.display ).toBe('none');
        
    });

    test('should show error on search batman123', () => {
        
        render(
            <MemoryRouter initialEntries={['/search?q=batman123']}>
                <SearchHeroPage />
            </MemoryRouter>
        );

        const alert = screen.getByLabelText('alert-danger');
        expect( alert.style.display ).toBe('');
        

    });

    test('should search correctly', () => {
        
        const inputValue = 'superman';

        render(
            <MemoryRouter initialEntries={['/search']}>
                <SearchHeroPage />
            </MemoryRouter>
        );

        const input = screen.getByRole('textbox');
        fireEvent.change( input, { target: { name: 'searchText', value: inputValue }})
        
        
        const form = screen.getByRole('form');
        fireEvent.submit( form );
        
        expect( mockedUseNavigate ).toHaveBeenCalledWith(`?q=${ inputValue }`)

    });


});