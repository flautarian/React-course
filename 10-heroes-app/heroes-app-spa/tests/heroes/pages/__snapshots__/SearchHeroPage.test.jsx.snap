// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchHeroPage test Should show default correctly 1`] = `
<div>
  <div
    class="animate__animated animate__fadeInDown"
  >
    <h1>
      SearchHeroPage
    </h1>
    <hr />
  </div>
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching...
      </h4>
      <hr />
      <form
        aria-label="form"
      >
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-primary mt-2"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results...
      </h4>
      <hr />
      <div
        aria-label="alert-danger"
        class="alert alert-danger animate__animated animate__fadeIn"
        style="display: none;"
      >
        No hero with 
        <b />
      </div>
    </div>
  </div>
</div>
`;

exports[`SearchHeroPage test should show default correctly 1`] = `
<div>
  <div
    class="animate__animated animate__fadeInDown"
  >
    <h1>
      SearchHeroPage
    </h1>
    <hr />
  </div>
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching...
      </h4>
      <hr />
      <form
        aria-label="form"
      >
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-primary mt-2"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results...
      </h4>
      <hr />
      <div
        aria-label="alert-danger"
        class="alert alert-danger animate__animated animate__fadeIn"
        style="display: none;"
      >
        No hero with 
        <b />
      </div>
    </div>
  </div>
</div>
`;
